.github/worksflows/pr-internal.yml

```diff
@@ -0,0 +1,89 @@
+name: Python SDK Conditional Workflow
+
+on:
+  pull_request:
+    branches:
+      - main
+      - dev
+  push:
+    branches:
+      - dev
+
+jobs:
+  check_branch_protection:
+    if: github.event_name == 'pull_request' && github.base_ref == 'main'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check source branch
+        run: |
+          if [[ "${{ github.head_ref }}" != "dev" ]]; then
+            echo "❌ Pull requests to main branch must come from dev branch"
+            echo "Current source branch: ${{ github.head_ref }}"
+            exit 1
+          fi
+          echo "✅ Source branch check passed"
+
+  Timestamp:
+    if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
+    needs: [check_branch_protection]
+    uses: storyprotocol/gha-workflows/.github/workflows/reusable-timestamp.yml@main
+
+  pr_build_and_test:
+    if: github.event_name == 'pull_request'
+    needs: [Timestamp]
+    uses: storyprotocol/gha-workflows/.github/workflows/reusable-build-unit-test-workflow.yml@main
+    with:
+      sha: ${{ github.event.pull_request.head.sha }}
+      ENVIRONMENT: "aeneid"
+    secrets:
+      WALLET_PRIVATE_KEY: ${{ secrets.WALLET_PRIVATE_KEY }}
+      RPC_PROVIDER_URL: ${{ secrets.RPC_PROVIDER_URL }}
+
+  push_build_and_test:
+    if: github.event_name == 'push'
```

Comment:
Should we do integration test in pull_request too? Do we run the test in Aeneid? Also is this reusable workflow for python test?